maestrowf/datastructures/core/executiongraph.py

```diff
@@ -645,7 +645,7 @@ def execute_ready_steps(self):
             # Just return for now, but we'll need a way to signal that there
             # are no more things to run.
             logging.info("'%s' is complete. Returning.", self.name)
-            return True
+            return StudyStatus.FINISHED
```

Comment:
@kcathey and I were talking about this condition. I found that the conductor returned 0 (FINISHED) status even when cancelled. @kcathey pointed out that this conditional gets triggered without checking for cancelled state. Kevin is going to look into a refactor to clean this up because I had initially done this to assess the state entering and leaving the status check to avoid needlessly waiting for another round of updates.